addons/l10n_id_efaktur/tests/test_l10n_id_efaktur.py

```diff
@@ -277,3 +279,304 @@ def test_efaktur_consume_code(self):
+            "invoice_line_ids": [
+                Command.create({"name": "line1", "price_unit": 110.0, "tax_ids": self.tax_id.ids}),
+            ],
+            "l10n_id_kode_transaksi": "01",
+        })
+        out_invoice_2.action_post()
+
+        # Generate efaktur separately between 1 and 2
+        out_invoice_1.download_efaktur()
+        out_invoice_2.download_efaktur()
+
+        self.assertIsNotNone(out_invoice_1.l10n_id_efaktur_document)
+        self.assertIsNotNone(out_invoice_2.l10n_id_efaktur_document)
+        self.assertNotEqual(out_invoice_1.l10n_id_efaktur_document.id, out_invoice_2.l10n_id_efaktur_document.id)
+
+        # if we generate an efaktur for the 2 together, it will raise an error
+        with self.assertRaises(UserError):
+            (out_invoice_1 + out_invoice_2).download_efaktur()
+
+        # even if cancel only one invoice, when generate efaktur together should still raise an error
+        empty_recordset_document = self.env['l10n_id_efaktur.document']
+
+        out_invoice_1.button_cancel()
+        out_invoice_1.button_draft()
+        out_invoice_1.action_post()
+        self.assertEqual(out_invoice_1.l10n_id_efaktur_document, empty_recordset_document)
+        with self.assertRaises(UserError):
+            (out_invoice_1 + out_invoice_2).download_efaktur()
+
+        # if we cancel invoice 2, then we download efaktur together, they should link to the same document
+        out_invoice_2.button_cancel()
+        out_invoice_2.button_draft()
+        out_invoice_2.action_post()
+        self.assertEqual(out_invoice_2.l10n_id_efaktur_document, empty_recordset_document)
+
+        (out_invoice_1 + out_invoice_2).download_efaktur()
+
+        self.assertEqual(out_invoice_1.l10n_id_efaktur_document.id, out_invoice_2.l10n_id_efaktur_document.id)
+
+        # if we try to generate again for same set of invoice, should link to same document (and attachmetn)
+        document = out_invoice_1.l10n_id_efaktur_document
+
+        (out_invoice_1 + out_invoice_2).download_efaktur()
+        self.assertEqual(out_invoice_1.l10n_id_efaktur_document.id, document.id)
+        self.assertEqual(out_invoice_2.l10n_id_efaktur_document.id, document.id)
+
+        # If we cancel one of the 2 invoices, should unset document for both
+        out_invoice_1.button_draft()
+        out_invoice_1.button_cancel()
+
+        self.assertEqual(out_invoice_1.l10n_id_efaktur_document, empty_recordset_document)
+        self.assertEqual(out_invoice_2.l10n_id_efaktur_document, empty_recordset_document)
+
+        # also ensure that message is logged for both invoices
+        self.assertEqual(
+            out_invoice_1.message_ids[0].body,
+            Markup("<p>e-Faktur document removed due to invoice cancellation</p>")
+        )
+
+        # check on the download result
+        action = (out_invoice_1 + out_invoice_2).download_efaktur()
+
+        self.authenticate(self.env.user.login, self.env.user.login)
```

Comment:
```suggestion
```

This one is not needed in this test I think

Reply:
without the authentication it seems like there's no content response

Reply:
Keep it then, I'm not very familiar with this flow and though it would be authenticated as self.env.user by default.